sympy/parsing/sympy_parser.py

```diff
@@ -1087,7 +1081,7 @@ def parse_expr(s, local_dict=None, transformations=standard_transformations,
                     a transformation should be function that
                     takes 3 arguments'''))
 
-    code = stringify_expr(s, local_dict, global_dict, transformations)
+    code = stringify_expr(s, local_dict, global_dict, transformations)  # type: ignore
```

Comment:
I agree it's not helpful to allow it to be None, as `()` has the same effect. Changing its type signature to `tUnion[tuple, str]` won't break downstream code as long as `transformations = transformations or ()` is kept.

> Can mypy not see that?

As always it can't do this kind of type narrowing now:
```
sympy/parsing/sympy_parser.py:1084: error: Argument 4 to "stringify_expr" has incompatible type "Union[Tuple[Callable[[List[Tuple[int, str]], Dict[str, Any], Dict[str, Any]], List[Tuple[int, str]]], ...], str]"; expected "Tuple[Callable[[List[Tuple[int, str]], Dict[str, Any], Dict[str, Any]], List[Tuple[int, str]]], ...]"
```